### ISSUE TITLE
Is the dataset actually correctly sharded in DDP finetuning training?

### ISSUE THREAD

**Issue by David0tt** [openvla/openvla#323]
Hello,

I am trying to run the finetuning using multiple gpus, so I am using a command roughly like this:

```
torchrun --standalone --nnodes 1 --nproc-per-node 4 vla-scripts/finetune.py \
  --vla_path "openvla/openvla-7b" \
  --data_root_dir /datasets \
  --dataset_name bridge_orig \
  --lora_rank 32 \
  --batch_size 16 \
  --grad_accumulation_steps 1 \
  --learning_rate 5e-4 \
  --image_aug True \
  --save_steps 2000
```

Expected behavior would be, for the training to take the same time (since we are running for a total of max_steps), one step roughly taking the same time (up to some overhead for communication), but the convergence behavior should be 4 times as fast (since we now have an accumulated gradient over batch_size * nproc-per-node = 16 * 4 = 64, instead of the 16 when using single-gpu training). It is hard to actually evaluate the convergence, since the wandb output graphs are very noisy. 

However, this only works, if the different gpus actually get different subsets of the training data (i.e. sharding). I see no obvious way, how this sharding is happening in the code. In particular, I would expect something like this should be added to the end of `class RLDSDataset(IterableDataset):` at the end of `__init__` (in `prismatic.vla.datasets.datasets.py`)

```
if dist.is_initialized() and dist.get_world_size() > 1:
  rank = dist.get_rank()
  world_size = dist.get_world_size()
  self.dataset = self.dataset.shard(num_shards=world_size, index=rank)
  self.dataset_length = self.dataset_length // world_size
```

However, i did some testing, printing out parts of the first batches being produced in each process (before manually adding sharding), and it appears that they are different. So I have two hyptohesis:
- Either RLDSDataset / dlimp already implements some kind of sharding
- Or RLDSDataset / dlimp is loaded with a different random seed in each process, resulting in obtaining a different random batch in each process (In this case, we get roughly the same result as in sharding, but now all 4 processes go through the whole randomly shuffled dataset independently, compared to in sharding where each of the four processes would go through 1/4 of the non-overlapping dataset).

Can you provide me with some insight, whether how I understand it is correct?
